osu.Game.Rulesets.Taiko/Edit/TaikoSelectionHandler.cs

```diff
@@ -69,7 +69,11 @@ public void SetRimState(bool state)
         {
             EditorBeatmap.PerformOnSelection(h =>
             {
-                if (h is Hit taikoHit) taikoHit.Type = state ? HitType.Rim : HitType.Centre;
+                if (h is Hit taikoHit)
+                {
+                    taikoHit.Type = state ? HitType.Rim : HitType.Centre;
+                    EditorBeatmap.Update(h);
```

Comment:
I've added these calls in line to match the rest, but we can potentially decide that it should be implicitly run on all items in the selection. This would be safer, but potentially increase the overhead when not all items in the selection are actually being modified (though i believe this is quite rare).